plugins/woocommerce/src/Internal/BatchProcessing/BatchProcessingController.php

```diff
@@ -528,7 +528,20 @@ private function log_consistent_failure( BatchProcessorInterface $batch_processo
 	 * @since 9.1.0
 	 */
 	private function remove_or_retry_failed_processors(): void {
-		if ( as_has_scheduled_action( self::WATCHDOG_ACTION_NAME ) ) {
+		if ( ! did_action( 'wp_loaded' ) ) {
+			return;
+		}
+
+		$last_error = error_get_last();
+		if ( ! is_null( $last_error ) && in_array( $last_error['type'], array( E_ERROR, E_PARSE, E_CORE_ERROR, E_COMPILE_ERROR, E_USER_ERROR, E_RECOVERABLE_ERROR ), true ) ) {
```

Comment:
There's [another place](https://github.com/woocommerce/woocommerce/blob/a93343dd90aa16c3182a7baf21d6be0d5e837e89/plugins/woocommerce/includes/class-woocommerce.php#L336) where we make basically this same kind of check, with the array of different error types. I wonder if it would be worth abstracting this so we can be consistent with the types of errors that qualify? Perhaps out of scope for this PR.

Reply:
Agree! I [logged a housekeeping task](https://github.com/woocommerce/woocommerce/issues/49245) (just to minimize delay, since we have some time constraints for this PR).